contracts/Stake.vy

```diff
@@ -15,35 +15,35 @@ stakeDates: HashMap [address, uint256]
 wolvercoinContract: Token
 activeUserContract: ActiveUser
 
+@external 
+def __init__(_wolvercoinContract: Token, _activeUserContract: ActiveUser):
+    self.wolvercoinContract = _wolvercoinContract
+    self.activeUserContract = _activeUserContract
+
 @external
 def stake (user: address, amountStaked: uint256):
     assert self.activeUserContract.getActiveUser (user)
     self.stakeAmounts[user] += amountStaked
     self.stakeDates[user] = block.timestamp
     self.wolvercoinContract.transferFrom (user, self, amountStaked)
 
-@external 
-def __init__(_wolvercoinContract: Token, _activeUserContract: ActiveUser):
-    self.wolvercoinContract = _wolvercoinContract
-    self.activeUserContract = _activeUserContract
-
 @external
 def unstake (_userAddress: address, amtUnstaked: uint256):
-    assert amtUnstaked < self.stakeAmounts[_userAddress]
-
+    assert amtUnstaked <= self.stakeAmounts[_userAddress]
     changeInTime: uint256 = block.timestamp - self.stakeDates[_userAddress]
     if changeInTime < 1210000:
         decimalAmt: decimal = convert (amtUnstaked, decimal)
         oneThird: decimal = decimalAmt / 3.0
         newAmt: uint256 = 2 * convert(oneThird, uint256)
         self.wolvercoinContract.transferFrom (self, _userAddress, newAmt)
-        self.wolvercoinContract.burnFrom (self, convert (oneThird, uint256))
         self.stakeAmounts[_userAddress] -= amtUnstaked
     else:
         days: uint256 = changeInTime/86400
-        percent: decimal = (convert(101**days, decimal) / convert(100**days, decimal))
-        newAmt: uint256 = amtUnstaked * (convert((percent),uint256))
+        percent: uint256 = 3800 / 365
```

Comment:
Should this be a `decimal` type instead of a `uint256` type for precision? And then you can cast the whole thing _after_ multiplying

Reply:
I'm keeping this as a uint256 and adjusting the decimal value manually in the actual calculation for newAmt